Core/include/Acts/TrackFitting/GlobalChiSquareFitter.hpp

```diff
@@ -709,8 +709,8 @@ class Gx2Fitter {
     // new track and delete it after updating the parameters. However, if we
     // would work on the externally provided track container, it would be
     // difficult to remove the correct track, if it contains more than one.
-    Acts::VectorTrackContainer trackContainerTempBackend;
-    Acts::VectorMultiTrajectory trajectoryTempBackend;
+    track_container_t trackContainerTempBackend;
+    traj_t trajectoryTempBackend;
```

Comment:
Wait, wouldn't this be the one that **can** be different from the outer backend? Especially if you're going to reuse this container, it does actually make sense to constrain it to the transient one.

Reply:
Yes, in theory it could be different, but in then end it would require setting up the config for the propagation a second time templated on vector.

Reply:
*typed

Reply:
Rolling it up from the end, looking at our temporary propagation:

We propagate and safe the states in `Gx2FitterResult`:
```
auto& r = propagatorState.template get<Gx2FitterResult<traj_t>>();
r.fittedStates = &trajectoryTempBackend;
```
This one we could either template on the user input `traj_t` or the agnostic `Acts::VectorMultiTrajectory`.

For the propagation, we also use the `Actor`, which is part of the whole `Gx2Fitter`, which again is templated on `traj_t`.

There, we would get a mismatch between.

Reply:
I will try to template the Actor more and set it up twice, then we can keep the `Acts::VectorMultiTrajectory` for now.

Reply:
I would just copy the track states explicitly from the temporary to the result container

Reply:
As I understood, the `Actor` can not run, since we use `Gx2FitterResult<traj_t>` for returning the result.
The temporary track states are discarded completely after fitting. So we are not using them for the final container.

Reply:
as just discussed with @andiwand:

I tried to template through the trajectory, but in the end we will get to a problem with
```
gx2fActor.extensions = gx2fOptions.extensions;
```
since the extensions are user-provided and also templated on `traj_t`.

For now the proposed option should work.